File: Apricot/Sprite.cs
using System;
using System.Windows;
using System.Windows.Media.Imaging;

namespace Apricot
{
    public class Sprite
    {
        private string path = null;
        private Point location = new Point(0, 0);
        private Size size = Size.Empty;
        private double opacity = 1;

        public string Path
        {
            get
            {
                return this.path;
            }
            set
            {
                this.path = value;
            }
        }

        public Point Location
        {
            get
            {
                return this.location;
            }
            set
            {
                this.location = value;
            }
        }

        public Size Size
        {
            get
            {
                return this.size;
            }
            set
            {
                this.size = value;
            }
        }

        public double Opacity
        {
            get
            {
                return this.opacity;
            }
            set
            {
                this.opacity = value;
            }
        }

        public Sprite(string path)
        {
            this.path = path;
        }
    }
}

File: Apricot/Sprite.swift
//
//  Sprite.swift
//  Apricot
//
//  Created by Milch on 2023/07/01.
//

import Foundation

public struct Sprite: Equatable {
    public var path: String? = nil
    public var location: CGPoint = CGPoint.zero
    public var size: CGSize = CGSize.zero
    public var opacity: Double = 1.0
    public var delay: Double = 0.0
    
    public static func == (lhs: Sprite, rhs: Sprite) -> Bool {
        return lhs.path == rhs.path && lhs.location == rhs.location && lhs.size == rhs.size && lhs.opacity == rhs.opacity && lhs.delay == rhs.delay
    }
}
